bin/projects/dbatools/dbatools/Parameter/DbaInstanceParameter.cs

```diff
@@ -279,18 +279,32 @@ public DbaInstanceParameter(string Name)
             // Connection String interpretation
             try
             {
-                System.Data.SqlClient.SqlConnectionStringBuilder connectionString = new System.Data.SqlClient.SqlConnectionStringBuilder(tempString);
+                System.Data.SqlClient.SqlConnectionStringBuilder connectionString =
+                    new System.Data.SqlClient.SqlConnectionStringBuilder(tempString);
                 DbaInstanceParameter tempParam = new DbaInstanceParameter(connectionString.DataSource);
                 _ComputerName = tempParam.ComputerName;
-                if (tempParam.InstanceName != "MSSQLSERVER") { _InstanceName = tempParam.InstanceName; }
-                if (tempParam.Port != 1433) { _Port = tempParam.Port; }
+                if (tempParam.InstanceName != "MSSQLSERVER")
+                {
+                    _InstanceName = tempParam.InstanceName;
+                }
+                if (tempParam.Port != 1433)
+                {
+                    _Port = tempParam.Port;
+                }
                 _NetworkProtocol = tempParam.NetworkProtocol;
 
                 IsConnectionString = true;
 
                 return;
             }
-            catch { }
+            catch (ArgumentException ex)
+            {
+                //TODO: I guess this is a problem for non-english versions of windows.
+                if (ex.Message.StartsWith("Keyword not supported:"))
+                {
+                    throw;
+                }
+            }
```

Comment:
This has two issues:

 - It will only catch argument exceptions. All other exception types will kill conversion (I probably could live with that, since almost all exceptions should be argument exceptions)
 - Indeed it will only work on an English OS.

You can get language-neutral identification by processing the target site:
```c#
catch (ArgumentException ex)
{
    string name = "unknown";
    try
    {
        name = ex.TargetSite.GetParameters()[0].Name;
    }
    catch { }
    if (name == "keyword")
    {
        throw;
    }
}
catch
{

}
```